main.go

```diff
@@ -109,29 +123,14 @@ func main() {
 			xmetrics.NewRegistry,
 			xmetricshttp.Unmarshal("prometheus", promhttp.HandlerOpts{}),
 			xhealth.Unmarshal("health"),
-			func(v *viper.Viper, logger log.Logger) (*chrysom.ClientConfig, error) {
-				config := new(chrysom.ClientConfig)
-				err := v.UnmarshalKey("argus", &config)
-				config.Logger = logger
-				// TODO: What to do? This is a discard provider because we don't create providers in uber/fx style
-				config.MetricsProvider = provider.NewDiscardProvider()
-				logging.Info(logger).Log(logging.MessageKey(), fmt.Sprintf("address: %s", config.Address))
-				return config, err
-			},
+			newArgusClientConfig,
 			webhook.NewFactory,
 			func(lc fx.Lifecycle, factory *webhook.Factory, metrics webhook.WebhookMetrics) http.Handler {
-				webhookRegistry, webhookHandler := factory.NewRegistryAndHandler(metrics)
-				lc.Append(fx.Hook{
-					OnStop: func(ctx context.Context) error {
-						close(webhookRegistry.Changes)
```

Comment:
We're not the ones pushing changes to this channel so we shouldn't be closing it.